src/main/java/org/openrewrite/java/testing/mockito/PowerMockitoWhenNewToMockito.java

```diff
@@ -0,0 +1,141 @@
+
+                            Cursor containingMethod = getCursor().dropParentUntil(x -> x instanceof J.MethodDeclaration);
+                            Expression argument = select2.getArguments().get(0);
+                            if (argument instanceof J.FieldAccess) {
+                                ArrayList<J.FieldAccess> listOfMocks = containingMethod.getMessage("POWERMOCKITO_WHEN_NEW_REPLACED", new ArrayList<J.FieldAccess>());
+                                listOfMocks.add((J.FieldAccess) argument);
+                                containingMethod.putMessage("POWERMOCKITO_WHEN_NEW_REPLACED", listOfMocks);
+                                return null;
+                            }
+                        }
+                    }
+                }
+                return super.visitMethodInvocation(method, ctx);
+            }
+
+            @Override
+            public J visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                J ret = super.visitMethodDeclaration(method, ctx);
+                List<J.FieldAccess> mockArguments = getCursor().getMessage("POWERMOCKITO_WHEN_NEW_REPLACED");
+                if (mockArguments != null && ret instanceof J.MethodDeclaration) {
+                    doAfterVisit(removeMockUsagesVisitor(mockArguments, method.toString()));
+
+                    J.MethodDeclaration retM = (J.MethodDeclaration) ret;
+
+                    // onlyIfReferenced=false as `maybeAddImport` doesn't seem to find the type referred to in a try statement
+                    // see https://github.com/openrewrite/rewrite/issues/5187
+                    maybeAddImport("org.mockito.MockedConstruction", false);
+
+                    for (J.FieldAccess mockArgument: mockArguments) {
+                        String mockedClassName = ((J.Identifier) mockArgument.getTarget()).getSimpleName();
+                        String variableNameForMock = VariableNameUtils.generateVariableName("mock" + mockedClassName, updateCursor(ret), VariableNameUtils.GenerationStrategy.INCREMENT_NUMBER);
+                        JavaTemplate template = JavaTemplate.builder(String.format("try (MockedConstruction<%s> %s = Mockito.mockConstruction(%s.class)) { } ", mockedClassName, variableNameForMock, mockedClassName))
+                                .contextSensitive()
+                                .imports("org.mockito.MockedConstruction")
+                                .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "mockito-core"))
+                                .build();
+                        J.MethodDeclaration applied = template.apply(getCursor(), method.getCoordinates().replaceBody());
+                        J.Try tryy = (J.Try) applied.getBody().getStatements().get(0);
+                        retM = applied.withBody(applied.getBody().withStatements(Collections.singletonList(tryy.withBody(retM.getBody()))));
```

Comment:
I had a hard time following this. I would do it just a little differently:

```suggestion
                        J.MethodDeclaration applied = JavaTemplate.builder(String.format(
                                "try (MockedConstruction<%s> %s = Mockito.mockConstruction(%s.class)) { }", mockedClassName, variableNameForMock, mockedClassName))
                                .contextSensitive()
                                .imports("org.mockito.MockedConstruction")
                                .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "mockito-core"))
                                .build()
                                .apply(getCursor(), method.getCoordinates().replaceBody());
                        J.Try try_ = ((J.Try) applied.getBody().getStatements().get(0)).withBody(retM.getBody());
                        retM = applied.withBody(applied.getBody().withStatements(singletonList(try_)));
```

Reply:
f0f8a3f